packages/playwright-test/src/runner.ts

```diff
@@ -270,13 +271,41 @@ export class Runner {
       });
       filesByProject.set(project, testFiles);
     }
-    return { filesByProject, setupFiles };
+
+    let applyFilterToSetup = false;
+    if (testFileFilters.length && setupFiles.size) {
+      // If the filter didn't match any tests apply it to the setup files.
+      if (setupFiles.size === fileToProjectName.size) {
+        applyFilterToSetup = true;
+        for (const [project, files] of filesByProject) {
```

Comment:
```suggestion
        // We now have only setup files in filesByProject, because all test files were filtered out.
        for (const [project, setupFiles] of filesByProject) {
```